armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -387,6 +387,119 @@ def getNumberDensityWithTrace(component, nucName):
     return nvt, nv
 
 
+class CylindricalComponentsAverageBlockCollection(BlockCollection):
+    """
+    Creates a representative block for the purpose of cross section generation with a one-dimensional
+    cylindrical model.
+
+    Notes
+    -----
+    When generating the representative block within this collection, the geometry is checked
+    against all other blocks to ensure that the number of components are consistent. This implementation
+    is intended to be opinionated, so if a user attempts to put blocks that have geometric differences
+    then this will fail.
+
+    This selects a representative block based on the collection of candidates based on the
+    median block average temperatures as an assumption.
+    """
+
+    def _getNewBlock(self):
+        newBlock = copy.deepcopy(self._selectCandidateBlock())
+        newBlock.name = "1D_CYL_AVG_" + newBlock.getMicroSuffix()
+        return newBlock
+
+    def _selectCandidateBlock(self):
+        """Selects the candidate block with the median block-average temperature."""
+        info = []
+        for b in self.getCandidateBlocks():
+            info.append((b.getAverageTempInC() * self.getWeight(b), b.getName(), b))
+        info.sort()
+        medianBlockData = info[len(info) // 2]
+        return medianBlockData[-1]
+
+    def _makeRepresentativeBlock(self):
+        """Build a representative fuel block based on component number densities."""
+        repBlock = self._getNewBlock()
+        bWeights = [self.getWeight(b) for b in self.getCandidateBlocks()]
+        componentsInOrder = self._orderComponentsInGroup(repBlock)
+
+        for c, allSimilarComponents in zip(repBlock, componentsInOrder):
+            allNucsNames, densities = self._getAverageComponentNucs(
+                allSimilarComponents, bWeights
+            )
+            for nuc, aDensity in zip(allNucsNames, densities):
+                c.setNumberDensity(nuc, aDensity)
+        return repBlock
+
+    def _getNucTempHelper(self):
+        raise NotImplementedError
+
+    @staticmethod
+    def _getAllNucs(components):
+        """Iterate through components and get all unique nuclides."""
+        nucs = set()
+        for c in components:
+            nucs = nucs.union(c.getNuclides())
+        return sorted(list(nucs))
+
+    @staticmethod
+    def _checkComponentConsisteny(b, repBlock):
+        """
+        Verify that all components being homogenized are rectangular and have consistent dimensions.
```

Comment:
This function does not check the shape of a component, just multiplicity and nuclides.